### PR TITLE
Mark config-trusted-resources as deprecated

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit marks config-trusted-resources as deprecated, VerificationPolicy has covered the features of config-trusted-resources so there's no need to keep it.

/kind misc

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: config-trusted-resources of trusted resources to store public key is deprecated and will be removed in release 0.46, please use VerificationPolicy instead 
```

### CODE REVIEW

File: pkg/apis/config/trusted_resources.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 // TrustedResources holds the collection of configurations that we attach to contexts.
 // Configmap named with "config-trusted-resources" where cosign pub key path and
 // KMS pub key path can be configured
+// Deprecated.
```
Comment (JeromeJu) [inline pkg/apis/config/trusted_resources.go:32]: Maybe we would like to add 'Use verification policy instead' for pointing users to the alternative.

**Review by JeromeJu — APPROVED** [review]
Thanks Yongxuan

**Comment by JeromeJu** [comment]
Maybe we would like to add 'Use verification policy instead' for pointing users to the alternative.
^ I think this might also apply to the release note

File: pkg/apis/config/trusted_resources.go (hunk 1 above)
Comment (Yongxuanzhang) [reply pkg/apis/config/trusted_resources.go:32]: Sure! Thanks for the suggestion!

File: docs/deprecations.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ being deprecated.
...
 | [`pipelineRef.bundle` and `taskRef.bundle` are deprecated](https://github.com/tektoncd/pipeline/issues/5514)                                                                                                                 | v0.41.0                                                              | Alpha                                                                                                  | July 13, 2023                       |
-
+| [`config-trusted-resources` is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | April 10, 2023 or v0.47.0                       |
```
Comment (lbernick) [inline docs/deprecations.md:29]: Do we need to update the alpha features table to explain that this feature is going to be removed? I'm also confused what the "config-trusted-resources" feature actually is. Is that the feature enabled by the flag "resource-verification-mode"? How would someone use VerificationPolicy, if that's the replacement feature?
Lastly, why are we deciding to support this for two more months?

Comment (Yongxuanzhang) [reply docs/deprecations.md:29]: - This feature stores public keys in config map, it's a subset feature of VerificationPolicy. 
- I will update the docs as @JeromeJu suggested. 
- The announcement is 0.45, are we ok to remove it in 0.46? If so I think we could change it to 0.46

Comment (lbernick) [reply docs/deprecations.md:29]: sg, please lmk when you've updated the PR to make it more clear how to use the replacement feature. re: when can we remove it? we can remove alpha features at any time according to our compatibility policy, so I'd put N/A (alpha)

File: pkg/apis/config/trusted_resources.go (hunk 1 above)
Comment (lbernick) [inline pkg/apis/config/trusted_resources.go:32]: I'm surprised you don't need to re-run apigen here, what's going on?

File: docs/deprecations.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ being deprecated.
...
 | [`pipelineRef.bundle` and `taskRef.bundle` are deprecated](https://github.com/tektoncd/pipeline/issues/5514)                                                                                                                 | v0.41.0                                                              | Alpha                                                                                                  | July 13, 2023                       |
-
+| [`config-trusted-resources` is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | v0.46.0                       |
```
Comment (lbernick) [inline docs/deprecations.md:29]:
```suggestion
| [The `config-trusted-resources` configMap is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | v0.46.0                       |
```

**Review by lbernick — APPROVED** [review]

File: pkg/apis/config/trusted_resources.go (hunk 1 above)
Comment (Yongxuanzhang) [reply pkg/apis/config/trusted_resources.go:32]: I think this is config not api in v1beta1 or v1? when I run `update-openapigen.sh` nothing happens

**Comment by pritidesai** [comment]
please add a little more details in the release note, I do no think so but could anyone be using this configmap? Also, how about we also update the doc at https://github.com/tektoncd/pipeline/blob/main/docs/trusted-resources.md#config-key-at-configmap-will-be-deprecated?

**Comment by Yongxuanzhang** [comment]
> please add a little more details in the release note, I do no think so but could anyone be using this configmap? Also, how about we also update the doc at https://github.com/tektoncd/pipeline/blob/main/docs/trusted-resources.md#config-key-at-configmap-will-be-deprecated?

Thanks! I have updated the release notes. For the docs section, I think I have updated? What else am I missing?

File: pkg/apis/config/trusted_resources.go (hunk 1 above)
Comment (lbernick) [reply pkg/apis/config/trusted_resources.go:32]: Ah I see, we only generate deepcopy functions, thanks!

**Comment by XinruZhang** [comment]
It would be great if we can create an issue to track the removal of this ConfigMap :)
Thank you @Yongxuanzhang

**Pull request merged**